### ISSUE TITLE
Checkmatefile schema addition: ports for relations

### ISSUE THREAD

**Issue by larsbutler** [checkmate/chessboard#25]
Currently, the Checkmatefile schema allows for the definition of connections between services via well-known interfaces (e.g., http, mysql, https, etc.). These interfaces do not specify ports, which raises the question: How are ports resolved?

We could assume default ports for most things, but I think it would be useful and necessary to allow ports to be explicitly specified. For example, for an `http` connection, the default can just be port `80`, but we should allow Checkmatefile authors to any other port, e.g, `8080`. This feature would be especially useful for [custom components](https://github.com/checkmate/chessboard/issues/9) which happen to run one or more services on the same interface. For example, a component could, hypothetically, run two different http services, each on its own port. We would need to have a way to model this, and also to prevent ambiguity and conflicts.

**Comment by hhoover** [comment]
With docker the port can be dynamic, aka auto assigned by docker, and handled by sidekick containers/services. something to think about

**Comment by gondoi** [comment]
I agree we need to have an override option, or even random option as @hhoover mentioned, but we can also look to `/etc/services` for default ports if only the protocol is specified.

**Comment by larsbutler** [comment]
@gondoi Good call on `/etc/services`! It seems that Python's [socket module](https://docs.python.org/2/library/socket.html) can do this as well:

```python
>>> import socket
>>> socket.getservbyname('mysql')
3306
>>> socket.getservbyname('http')
80
>>> socket.getservbyname('https')
443
>>> socket.getservbyname('ssh')
22
>>> socket.getservbyname('postgresql')
5432
```
At least, it works for some things:
```python
>>> socket.getservbyname('mongodb')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.error: service/proto not found
>>> 
```
I suspect that this module is reading from /etc/services or has a pre-computed table with the same information.

**Comment by larsbutler** [comment]
@hhoover Regarding dynamic Docker ports: How does that work? All of the Docker examples I've seen performed explicit port mapping only. Link?

**Comment by hhoover** [comment]
Let's say you have multiple of the same container on the same host. They both can't listen on the same port on that host. So what happens is you don't define the host port, only the container port to be mapped. You then use a service discovery mechanism for keeping track of the ports for load balancing or linking containers. Example:

http://adetante.github.io/articles/service-discovery-with-docker-2/
